xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.cpp

```diff
@@ -148,6 +160,59 @@ void CVideoLayerBridgeDRMPRIME::Configure(CVideoBufferDRMPRIME* buffer)
     m_DRM->AddProperty(plane, "COLOR_ENCODING", GetColorEncoding(picture));
     m_DRM->AddProperty(plane, "COLOR_RANGE", GetColorRange(picture));
   }
+
+  struct connector* connector = m_DRM->GetConnector();
+  if (m_DRM->SupportsProperty(connector, "HDR_OUTPUT_METADATA"))
+  {
+    m_hdr_metadata.metadata_type = HDMI_STATIC_METADATA_TYPE1;
+    m_hdr_metadata.hdmi_metadata_type1 = {
+        .eotf = GetEOTF(picture),
+        .metadata_type = HDMI_STATIC_METADATA_TYPE1,
+    };
+
+    if (m_hdr_blob_id)
+      drmModeDestroyPropertyBlob(m_DRM->GetFileDescriptor(), m_hdr_blob_id);
+    m_hdr_blob_id = 0;
+
+    if (m_hdr_metadata.hdmi_metadata_type1.eotf)
+    {
+      const AVMasteringDisplayMetadata* mdmd = GetMasteringDisplayMetadata(picture);
+      if (mdmd && mdmd->has_primaries)
+      {
+        for (int i = 0; i < 3; i++)
+        {
+          m_hdr_metadata.hdmi_metadata_type1.display_primaries[i].x =
+              std::round(av_q2d(mdmd->display_primaries[i][0]) * 50000.0);
+          m_hdr_metadata.hdmi_metadata_type1.display_primaries[i].y =
+              std::round(av_q2d(mdmd->display_primaries[i][1]) * 50000.0);
+        }
+        m_hdr_metadata.hdmi_metadata_type1.white_point.x =
+            std::round(av_q2d(mdmd->white_point[0]) * 50000.0);
+        m_hdr_metadata.hdmi_metadata_type1.white_point.y =
+            std::round(av_q2d(mdmd->white_point[1]) * 50000.0);
+      }
+      if (mdmd && mdmd->has_luminance)
+      {
+        m_hdr_metadata.hdmi_metadata_type1.max_display_mastering_luminance =
+            std::round(av_q2d(mdmd->max_luminance));
+        m_hdr_metadata.hdmi_metadata_type1.min_display_mastering_luminance =
+            std::round(av_q2d(mdmd->min_luminance) * 10000.0);
+      }
+
```

Comment:
Would you mind to elaborate on the magic numbers used in this block?
I'd appreciate a comment if they come from some kernel sources or RFC's etc.
I'd guess that the `*10000.0` are the max nits defined in some HDR standard, but I haven't ever heard about the `50000` constant for example.

Reply:
The expected values are defined in CTA-861-G (6.9.1 Static Metadata Type 1), where `display_primaries` and `white_point` is

> coded as unsigned 16-bit values in units of `0.00002`, where `0x0000` represents
zero and `0xC350` represents `1.0000`

`max_display_mastering_luminance` is

> coded as an unsigned 16-bit value in units of `1 cd/m2`, where `0x0001` represents `1 cd/m2` and `0xFFFF` represents `65535 cd/m2`

and `min_display_mastering_luminance` is

> coded as an unsigned 16-bit value in units of `0.0001 cd/m2`, where `0x0001` represents `0.0001 cd/m2` and `0xFFFF` represents `6.5535 cd/m2`

The units of `0.00002` means we need to multiply the ffmpeg value (0.0-1.0) by `50000.0`, and same for units of `0.0001 cd/m2` we need to multiply by `10000.0` to get correct values expected by the kernel.

Reply:
Cool thx much for the explanation!!

Reply:
I have now added some comments in code regarding this conversion

Reply:
Thx much!